### PR TITLE
upgrade to 2.3.1

### PR DESCRIPTION
**Pull request by fogmann**
@angular/core: 2.3.1
    rxjs: 5.0.0-rc.4
    @angular/cli: 1.0.1

move app.module

### CODE REVIEW

**Comment by fogmann** [comment]
Trying to get  rxjs-socket.io to work in my project and it is not up-to-date.
Updated the files to make it current.

**Comment by thelgevold** [comment]
Thanks!
I was thinking it would make sense to update these dependencies to the latest versions. Angular 4.1 etc.

CLI is no longer in Beta, so it would be nice to bring that up to the latest as well.

File: client/src/app/chat/chat.component.ts [hunk 1]
```diff
@@ -3,10 +3,7 @@ import { ChatService }       from './chat.service';
 
 @Component({
   selector: 'chat-component',
-  template: `<div *ngFor="let message of messages">
-              {{message.text}}
-             </div>
-             <input [(ngModel)]="message" /><button (click)="sendMessage()">Send</button>`,
+  templateUrl: './chat.component.html',
```
Comment (thelgevold) [inline client/src/app/chat/chat.component.ts:6]: Is the html file missing? Can't see it in the diff?

**Comment by fogmann** [comment]
It is already there, but wasn't linked.

**Comment by thelgevold** [comment]
Thanks for the PR!

**Pull request merged**